modules/globebrowsing/src/rawtiledatareader.cpp

```diff
@@ -439,6 +442,94 @@ RawTileDataReader::~RawTileDataReader() {
     }
 }
 
+std::optional<std::string> RawTileDataReader::getMRFCache() {
+    GlobeBrowsingModule& module = *global::moduleEngine->module<GlobeBrowsingModule>();
+
+    std::string datasetIdentifier =
+        std::to_string(std::hash<std::string>{}(_datasetFilePath));
+    std::string path = fmt::format("{}/{}/{}/",
+        module.mrfCacheLocation(), _cacheProperties.path, datasetIdentifier);
+    std::string root = absPath(path).string();
+    std::string mrf = root + datasetIdentifier + ".mrf";
+    std::string cache = root + datasetIdentifier + ".mrfcache";
+
+    if (!std::filesystem::exists(mrf)) {
+        std::error_code ec;
+        if (!std::filesystem::create_directories(root, ec)) {
+            // Already existing directories causes a 'failure' but no error
+            if (ec) {
+                LWARNING(fmt::format(
+                    "Failed to create directories for cache at: {}. Error Code: {}, message: {}",
+                    root, std::to_string(ec.value()), ec.message()
+                ));
+                return std::nullopt;
+            }
+        }
+
+        GDALDriver* driver = GetGDALDriverManager()->GetDriverByName("MRF");
+        if (driver != nullptr) {
+            auto src = static_cast<GDALDataset*>(GDALOpen(_datasetFilePath.c_str(), GA_ReadOnly));
+            if (!src) {
```

Comment:
Can do `defer { GDALClose(src); }` to ensure the dataset is closed even in the cases that an `std::nullopt` is returned

Reply:
Nifty!